samples/deployment/aws/build_ami_vftd.yaml

```
@@ -0,0 +1,196 @@
+# Create Basic VM For AMI BUILD
+- name: "Build AMI image for vFTD"
+  hosts: localhost
+  remote_user: ubuntu
+  gather_facts: no
+  vars:
+    ftd_build_id: "6.4.0-1304"
+    ec2_instance_type: "c4.xlarge"
+    ec2_base_image: "ami-038f5168"
+    : "ngfwv-stage-temp-kp.pem"
+  tasks:
+    - name: "Download QCOW2"
+      get_url:
+        url: "https://...path.../...to.../{{ ftd_build_id }}.qcow2"
+        dest: "./"
+
+    # Prepare Env Section
+    - name: "Create an AWS EC2 key pair"
+      ec2_key:
+        name: "{{ ansible_ssh_private_key_file }}"
+      register: stage_kp
+    - name: "Save AWS EC2 private key"
+      copy:
+        content: "{{ stage_kp.key.private_key }}"
+        dest: "{{ ansible_ssh_private_key_file }}"
+        mode: 0600
+      when: stage_kp.changed
+    - name: "Create a VPC"
+      ec2_vpc_net:
+        name: "Stage VPC"
+        state: present
+        cidr_block: 10.0.0.0/16
+      register: stage_vpc
+    - name: "Create an Internet Gateway"
+      ec2_vpc_igw:
+        vpc_id: "{{ stage_vpc.vpc.id }}"
+        state: present
+      register: stage_vpc_igw
+    - name: "Create a Subnet"
+      ec2_vpc_subnet:
+        state: present
+        vpc_id: "{{ stage_vpc.vpc.id }}"
+        cidr: 10.0.1.0/24
+      register: stage_subnet
+    - name: "Create a Routing table"
+      ec2_vpc_route_table:
+        vpc_id: "{{ stage_vpc.vpc.id }}"
+        subnets:
+          - "{{ stage_subnet.subnet.id }}"
+        routes:
+          - dest: 0.0.0.0/0
+            gateway_id: "{{ stage_vpc_igw.gateway_id }}"
+      register: stage_vpc_route_table
+    - name: "Create a Security group"
+      ec2_group:
+        name: "Stage VPC Security Group"
+        description: "Security group for Stage VPC"
+        vpc_id: "{{ stage_vpc.vpc.id }}"
+        state: present
+        rules:
+          - proto: all
+            cidr_ip: 0.0.0.0/0
+      register: stage_vpc_security_group
+
+    # Create VM Section
+    - name: "Create staging VM"
```

```suggestion
    - name: "Create a staging VM"
```
Again, any reason to use `staging` here? Usually, a staging environment is used to define a nearly exact replica of a production environment used for testing, but I'm not sure if it suits this sample.

---

The only purpose to create all resources described in the playbook - is to create QCOW based AMI. That's why, at the end of the playbook we have cleanup phase.